File: 0121-BestTimeToBuyAndSellStock/BestTimeToBuyAndSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min = prices[0];
        int diff = 0;
        
        for (int i = 0; i < prices.size(); i++){
            int temp = prices[i] - min;
            if (temp > diff){
                diff = temp;
            }
            
            if (prices[i] < min){
                min = prices[i];
            }
        }
        
        return diff;
    }
};

File: 121-BestTimeToBuyAndSellStock/BestTimeToBuyAndSellStock.py
# Python3
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 0: return 0
        
        minPrice = 10000
        max = 0
        
        for i in range(len(prices)):
            if prices[i] < minPrice:
                minPrice = prices[i]
            if (prices[i]-minPrice) > max:
                max = prices[i]-minPrice
                
        return max
        
